Sources/Sentry/SentryMetricProfiler.mm

```diff
@@ -81,6 +75,27 @@ @implementation SentryMetricReading
     }
     return @ { @"unit" : unit, @"values" : timestampNormalizedValues };
 }
+
+/**
+ * @return a dictionary containing all the metric values recorded during the continuous profile with
+ * absolute timestamps, or @c nil if there were no metrics recorded
+ */
+SentrySerializedMetricEntry *_Nullable serializeContinuousProfileMetricReadings(
+    NSArray<SentryMetricReading *> *readings, NSString *unit)
+{
+    const auto *serializedValues = [NSMutableArray<SentrySerializedMetricReading *> array];
+    [readings enumerateObjectsUsingBlock:^(
+        SentryMetricReading *_Nonnull reading, NSUInteger idx, BOOL *_Nonnull stop) {
+        const auto value = [NSMutableDictionary dictionary];
+        value[@"timestamp"] = @(reading.absoluteNSDateInterval);
+        value[@"value"] = reading.value;
+        [serializedValues addObject:value];
+    }];
+    if (serializedValues.count == 0) {
+        return nil;
+    }
```

Comment:
`l`: You could do an early return if `readings.count == 0`, instead of checking if `serializedValues.count == 0` here. Then, you would skip allocating an NSMutableArray in that case.